javalib/src/main/scala/java/util/concurrent/atomic/AtomicReferenceArray.scala

```diff
@@ -0,0 +1,43 @@
+package java.util.concurrent.atomic
+
+import scala.collection.mutable.ArrayBuffer
+
+class AtomicReferenceArray[E <: AnyRef](
+    length: Int) extends Serializable {
+
+  def this(array: Array[E]) = {
+    this(array.size)
+    inner = ArrayBuffer[E](array: _*)
+  }
+
+  private var inner: ArrayBuffer[E] = new ArrayBuffer[E](length)
+
+  final def length(): Int = inner.size
+
+  final def get(i: Int): E = inner(i)
+
+  final def set(i: Int, newValue: E): Unit =
+    inner.insert(i, newValue)
```

Comment:
This is not correct. It would resize the array. It's supposed to be simply `inner(i) = newValue`.